torch/distributions/kl.py

```diff
@@ -199,12 +199,29 @@ def _kl_binomial_binomial(p, q):
     # kullback-leibler-divergence-for-binomial-distributions-p-and-q
     if (p.total_count < q.total_count).any():
         raise NotImplementedError('KL between Binomials where q.total_count > p.total_count is not implemented')
-    kl = p.total_count * (p.probs * (p.logits - q.logits) + (-p.probs).log1p() - (-q.probs).log1p())
+    kl = p.total_count * (p.probs * (p.logits - q.logits) + p._log1pmprobs() - q._log1pmprobs())
     inf_idxs = p.total_count > q.total_count
     kl[inf_idxs] = _infinite_like(kl[inf_idxs])
     return kl
 
 
+@register_kl(Binomial, Poisson)
```

Comment:
Heterogeneous combinations were placed below. This section was for homogeneous combinations.